### PR TITLE
Properties- Class Exercise

### PR DESCRIPTION
**Pull request by RonAzar**

### CODE REVIEW

File: src/main/kotlin/Properties/Tamir/PropertiesTamirClassEx.kt [hunk 1]
```diff
@@ -0,0 +1,37 @@
+package Properties.Tamir
+import java.time.DayOfWeek
+
+//1. create class called Store that initlize by day and list of products
+data class Store(val day: DayOfWeek,val products: List<Product>)
+{
+    //2. add property to that indicate if the store is open the store is open all the day expect saturday
+    val shomeretShabat:Boolean get() {
+        when(day){
```
Comment (tamir-shteinberg1) [inline src/main/kotlin/Properties/Tamir/PropertiesTamirClassEx.kt:9]: when you have only 2 option is better use if/else then using when

File: src/main/kotlin/Properties/Tamir/PropertiesTamirClassEx.kt [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    //3. add property to that indicate number of product
+    val numberOfProducts:Int get() {
+        return products.size
+    }
+
+    //5. I want to count number of calling get receipts
+    var countGetReceipts:Int = 0
+
+    //4. add val that get receipts //no need to implement the method but its an heavy task
+    val getReceipts:List<String> by lazy {
```
Comment (tamir-shteinberg1) [inline src/main/kotlin/Properties/Tamir/PropertiesTamirClassEx.kt:24]: using lazy here is not will do the trick, 
I'l explain in the first time you call getReceipts you will increase the counter but you also going to set the value to listOf("Hello","Bob"). meaning in the next time you going to call to getReceipts you';; get the list witout increase the counter.
In order to solve it you can use get() that store function and not value.

File: src/main/kotlin/Properties/Tamir/PropertiesTamirClassEx.kt [hunk 3]
```diff
@@ -0,0 +1,37 @@
+package Properties.Tamir
+import java.time.DayOfWeek
+
+//1. create class called Store that initlize by day and list of products
+data class Store(val day: DayOfWeek,val products: List<Product>)
+{
+    //2. add property to that indicate if the store is open the store is open all the day expect saturday
+    val shomeretShabat:Boolean get() {
```
Comment (tamir-shteinberg1) [inline src/main/kotlin/Properties/Tamir/PropertiesTamirClassEx.kt:8]: space after :, its not critical but its important to keep on code style

**Review by tamir-shteinberg1 — COMMENTED** [review]
Please use CMD + option + L to keep on code style, you have some cases the you need to add space

File: src/main/kotlin/Properties/Tamir/PropertiesTamirClassEx.kt (hunk 1 above)
Comment (RonAzar) [reply src/main/kotlin/Properties/Tamir/PropertiesTamirClassEx.kt:9]: Got it
Thanks 
fix it right now

File: src/main/kotlin/Properties/Tamir/PropertiesTamirClassEx.kt (hunk 2 above)
Comment (RonAzar) [reply src/main/kotlin/Properties/Tamir/PropertiesTamirClassEx.kt:24]: Got it 
thanks

File: src/main/kotlin/Properties/Tamir/PropertiesTamirClassEx.kt (hunk 3 above)
Comment (RonAzar) [reply src/main/kotlin/Properties/Tamir/PropertiesTamirClassEx.kt:8]: Fix it right now
Thanks for the review Tamir!

**Pull request merged**